MobiusCore/Source/Disposables/Task+Disposable.swift

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019-2024 Spotify AB.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public extension Task {
+
+    /// A disposable for use with `EffectHandler` that will cancel the task
+    ///
+    ///     func handle(_ parameters: Void, _ callback: EffectCallback<Event>) -> Disposable {
+    ///         Task {
+    ///
+    ///         }
+    ///        .asDisposable
+    ///     }
+    var asDisposable: any Disposable {
```

Comment:
Existential types are more expensive than concrete types, so it's better to avoid them if possible. In this case it should be easily avoidable by using `some` instead of `any`.

Reply:
Cool, when compiled, that's the same thing as `var asDisposable: AnonymousDisposable`, correct?

Reply:
Exactly, you could therefore also return just `AnonymousDisposable`. But I think that here the fact that the `Disposable` is of type `AnonymousDisposable` is insignificant from the API perspective, so it might be better to use the opaque return type (`some`).

Reply:
Agreed